app/main.py

```
@@ -1,3 +1,50 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError(f"Unsupported operand type for +: {type(other)}")
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
```

This method silently ignores unsupported types for `other`. For example, `Distance(10) += 'a'` would do nothing. This is inconsistent with your `__add__` method, which correctly raises a `TypeError` for unsupported types. For consistent and predictable behavior, consider adding a check to raise a `TypeError` here as well.